src/Estimators/src/BaseEstimatorFromFootIMU.cpp

```diff
@@ -170,100 +209,254 @@ bool BaseEstimatorFromFootIMU::advance()
+        m_measuredRotation = m_input.measuredRotation_L.rotation();
+        m_measuredAngularVelocity = m_input.measuredAngularVelocity_L;
+        // has the stance foot just changed?
+        if (m_isLastStanceFoot_R)
+        {
+            // updating the yawOld value.
+            Eigen::Vector3d lastRPY_R = toXYZ(m_state.footPose_R.rotation());
+            // Eigen::Vector3d lastRPY_L = toXYZ(m_input.measuredRotation_L.rotation());
+            m_yawOld = lastRPY_R(2);
+
+            // updating the m_T_walk matrix.
+            Eigen::Vector3d stepTras = (m_state.footPose_L.translation() - m_state.footPose_R.translation());
+            stepTras(2) = 0.0;
+            manif::SE3d T_walkTras(stepTras, manif::SO3d::Identity());
+
+            // Eigen::Matrix3d stepRot = m_state.footPose_R.rotation().inverse() * m_state.footPose_L.rotation();
+            // Eigen::Vector3d stepRotRPY = toXYZ(stepRot);
+            // manif::SO3d stepRotManif = manif::SO3d(stepRotRPY(0), stepRotRPY(1), stepRotRPY(2));
+            // manif::SE3d T_walkRot(m_noTras, stepRotManif);
+            // manif::SE3d temp = T_walkTras * T_walkRot * m_T_walk * m_T_yawDrift;
+
+            manif::SE3d temp = T_walkTras * m_T_walk;
+            m_T_walk = temp;
+
+            // m_T_yawDrift.setIdentity();
+        }
+    }
+
+    if (m_input.isRightStance)
+    {
+        stanceFootFrame_H_link = m_footFrame_H_link_R;
+        stanceFootFrameIndex = m_footFrameIndex_R;
+        // casting the measured foot orientation manif::SO3d --> Eigen::Matrix3d.
+        m_measuredRotation = m_input.measuredRotation_R.rotation();
+        m_measuredAngularVelocity = m_input.measuredAngularVelocity_R;
+        // has the stance foot just changed?
+        if (m_isLastStanceFoot_L)
+        {
+            // updating the yawOld value.
+            Eigen::Vector3d lastRPY_L = toXYZ(m_state.footPose_L.rotation());
+            // Eigen::Vector3d lastRPY_R = toXYZ(m_input.measuredRotation_R.rotation());
+            m_yawOld = lastRPY_L(2);
+
+            // updating the m_T_walk matrix.
+            Eigen::Vector3d stepTras = (m_state.footPose_R.translation() - m_state.footPose_L.translation());
+            stepTras(2) = 0.0;
+            manif::SE3d T_walkTras(stepTras, manif::SO3d::Identity());
+
+            // Eigen::Matrix3d stepRot = m_state.footPose_L.rotation().inverse() * m_state.footPose_R.rotation();
+            // Eigen::Vector3d stepRotRPY = toXYZ(stepRot);
+            // manif::SO3d stepRotManif = manif::SO3d(stepRotRPY(0), stepRotRPY(1), stepRotRPY(2));
+            // manif::SE3d T_walkRot(m_noTras, stepRotManif);
+            // manif::SE3d temp = T_walkTras * T_walkRot * m_T_walk * m_T_yawDrift;
+
+            manif::SE3d temp = T_walkTras * m_T_walk;
+            m_T_walk = temp;
+
+            // m_T_yawDrift.setIdentity();
+        }
+    }
+
+    m_stanceLinkIndex = m_model.getFrameLink(stanceFootFrameIndex);
```

Comment:
What if both ``isLeftStance`` and ``isRightStance`` are false? Can it happen? Maybe it is worth printing an error in this case?